noir-projects/aztec-nr/aztec/src/state_vars/shared_mutable/shared_mutable.nr

```diff
@@ -24,87 +31,150 @@ impl<T, DELAY> Storage<T> for SharedMutable<T, DELAY> {}
 // the value is not changed immediately but rather a value change is scheduled to happen in the future after some delay
 // measured in blocks. Reads in private are only valid as long as they are included in a block not too far into the 
 // future, so that they can guarantee the value will not have possibly changed by then (because of the delay).
-impl<T, DELAY> SharedMutable<T, DELAY> {
+// The delay for changing a value is initially equal to INITIAL_DELAY, but can be changed by calling 
+// `schedule_delay_change`.
+impl<T, INITIAL_DELAY> SharedMutable<T, INITIAL_DELAY> {
     pub fn new(context: Context, storage_slot: Field) -> Self {
         assert(storage_slot != 0, "Storage slot 0 not allowed. Storage slots must start from 1.");
-        Self { context, storage_slot, _dummy: [0; DELAY] }
+        Self { context, storage_slot }
     }
 
     pub fn schedule_value_change(self, new_value: T) {
```

Comment:
Would be neat with a small comment on this just to say it should only be called from public. You see it when reading it and it is unwrapping public context, just to make it more obvious. Same for a few of the functions, when we don't have the `#[aztec(public)]` and its family in is not as easy to follow.

Reply:
Eventually this will be indicated by #5886. More concretely though, do you think we should have both source and docsite documentation for these?

Reply:
Handling it with 5886 probably the better option all together 👍